homeassistant/components/shelly/__init__.py

```diff
@@ -150,13 +151,28 @@ def _async_input_events_handler(self):
             if (
                 "inputEvent" not in block.sensor_ids
                 or "inputEventCnt" not in block.sensor_ids
+                or "input" not in block.sensor_ids
             ):
                 continue
 
             channel = int(block.channel or 0) + 1
+            input_value = block.input
             event_type = block.inputEvent
             last_event_count = self._last_input_events_count.get(channel)
+            last_input_value = self._last_input_value.get(channel)
             self._last_input_events_count[channel] = block.inputEventCnt
+            self._last_input_value[channel] = input_value
+
+            if last_input_value is not None and input_value != last_input_value:
+                self.hass.bus.async_fire(
+                    "shelly.input_changed",
```

Comment:
We should only fire these events if the device is in detached state.

Reply:
Hi @bramkragten,
If you set the button type to any type which is not momentary a binary sensor will be created which reflects the input state.

Reply:
Ah! I see it now. It is disabled by default, that's why I missed it 🤦

Reply:
We need to spend some time on expanding the integration docs :-)